students/seungwonburm/users/views.py

```diff
@@ -1,39 +1,63 @@
-            email           = data['email']
-            password        = data['password']
-            nickname        = data['nickname']
-            phone_number    = data['phone_number']
+            data         = json.loads(request.body)
+            email        = data['email']
+            password     = data['password']
+            nickname     = data['nickname']
+            phone_number = data['phone_number']
 
-            email_regex     = "(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
-            password_regex  = "^(?=.*).{8,}$"
+            EMAIL_REGEX    = "(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
+            PASSWORD_REGEX = "^(?=.*).{8,}$"
 
-            if not re.search(password_regex, password):
+            if not re.search(PASSWORD_REGEX, password):
                 return JsonResponse({'message' : 'VALIDATION ERROR : INVALID PASSWORD'}, status=400)
-            if not re.search(email_regex, email):
+            if not re.search(EMAIL_REGEX, email):
                 return JsonResponse({'message' : 'VALIDATION ERROR : INVALID EMAIL'}, status=400)
             if Account.objects.filter(Q(email=email) | Q(nickname=nickname) | Q(phone_number=phone_number)).exists():
-                return JsonResponse({'message' : 'USER INFORMATION ALREADY EXISTS'}, status=400)
+                return JsonResponse({'message' : 'USER INFORMATION ALREADY EXISTS'}, status=409)
+            
+            hashed_password=bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt())
 
             Account.objects.create(
-                email           = email,
-                password        = password,
-                nickname        = nickname,
-                phone_number    = phone_number
+                email        = email,
+                password     = hashed_password.decode('utf-8'),
+                nickname     = nickname,
+                phone_number = phone_number
             )
             return JsonResponse({'message' : 'SUCCESS!'}, status=201)
+
         except KeyError:
             return JsonResponse({'message' : 'KEY_ERROR'}, status=400)
 
+class SigninView(View):
+    def post(self, request):
+        try:
+            data           = json.loads(request.body)
+            email          = data['email']
+            input_password = data['password']
+            
+            
+            if not email or not input_password:
+                return JsonResponse({'message' : 'KEY ERROR'}, status=400)
+            if not Account.objects.filter(email=email).exists(): 
+                return JsonResponse({'message' : 'INVALID USER'}, status=401)
+            user = Account.objects.get(email = email)
+            if not bcrypt.checkpw(input_password.encode('utf-8'), user.password.encode('utf-8')):
+                return JsonResponse({'message' : 'INVALID PASSWORD'}, status=401)
+            email_data = {'email': email}
+            token      = jwt.encode(email_data, SECRET_KEY, ALGORITHM)
```

Comment:
이정도면 굳이 변수로 dictionary 할당할 필요없이 아래처럼 한번에 처리해주시는게 더 깔끔할 것 같네요.
```suggestion
            token      = jwt.encode({'email': email}, SECRET_KEY, ALGORITHM)
```

Reply:
넵 메모리 관리 유의 하겠습니다.